File: rust/src/models/product_event.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{event_id::EventId, event_type::EventType, product_event_data::ProductEventData};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ProductEvent {
    #[serde(flatten)]
    pub flatten_producteventdata: ProductEventData,

    pub id: EventId,

    pub timestamp: String,

    pub r#type: EventType,
}

impl ProductEvent {
    pub fn new(id: EventId, timestamp: String, r#type: EventType) -> Self {
        Self {
            flatten_producteventdata: Default::default(),
            id,
            timestamp,
            r#type,
        }
    }
}

File: typescript/src/models/productEvent.ts
// this file is @generated
import { parseDateTime } from "../datetime";
import { type EventId, EventIdSerializer } from "./eventId";
import { type EventType, EventTypeSerializer } from "./eventType";
import { type ProductEventData, ProductEventDataSerializer } from "./productEventData";

export interface ProductEvent {
  flattenProducteventdata: ProductEventData;

  id: EventId;

  timestamp: Date;

  type: EventType;
}

export const ProductEventSerializer = {
  _fromJsonObject(object: any): ProductEvent {
    return {
      flattenProducteventdata: ProductEventDataSerializer._fromJsonObject(
        object["__flatten_producteventdata"]
      ),
      id: EventIdSerializer._fromJsonObject(object["id"]),
      timestamp: parseDateTime(object["timestamp"]),
      type: EventTypeSerializer._fromJsonObject(object["type"]),
    };
  },

  _toJsonObject(self: ProductEvent): any {
    return {
      __flatten_producteventdata: ProductEventDataSerializer._toJsonObject(
        self.flattenProducteventdata
      ),
      id: EventIdSerializer._toJsonObject(self.id),
      timestamp: self.timestamp,
      type: EventTypeSerializer._toJsonObject(self.type),
    };
  },
};
